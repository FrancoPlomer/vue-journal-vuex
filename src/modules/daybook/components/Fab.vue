<template>
    <button 
        class="btn btn-primary"
        @click="$emit('on:click')"
    >
        <i :class="icon"></i>
    </button>
</template>

<script>
export default {
    name: 'fab-buttom',
    props: {
        icon: {
            type: String,
            default: 'fa fa-2x fa-plus'
        }
    }

}
</script>

<style lang="scss" scoped>

    button {
        border-radius: 100%;
        bottom: 20px;
        height: 80px;
        position: fixed;
        right: 20px;
        width: 80px;
    }
</style>