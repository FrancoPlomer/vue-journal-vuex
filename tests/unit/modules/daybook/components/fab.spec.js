import { shallowMount } from "@vue/test-utils";

import Fab from "@/modules/daybook/components/Fab.vue";

describe('Fab', () => {
    test('Debe hacer match con el snapshot', () => {
        const wrapper = shallowMount( Fab )
        expect( wrapper.html() ).toMatchSnapshot()
    })
    test('Debe mostrar icono por defecto', () => {
        const wrapper = shallowMount( Fab )
        const icon = wrapper.find('i')
        expect(icon.attributes().class).toBe('fa fa-2x fa-plus')
    })
    test('Debe mostrar icono por argumento: fa-circle', () => {        
        const iconProp = 'fa fa-2x fa-circle'
        const wrapper = shallowMount( Fab, {
            props: {
                icon: iconProp
            }
        })
        const icon = wrapper.find('i')
        expect(icon.attributes().class).toBe('fa fa-2x fa-circle')

    })
    test('Debe de emitir el evento on:click cuando se hace click', async() => {
        const wrapper = shallowMount( Fab )
        await wrapper.find('i').trigger('click')
        expect(wrapper.emitted('on:click')).toBeTruthy()
    })
})