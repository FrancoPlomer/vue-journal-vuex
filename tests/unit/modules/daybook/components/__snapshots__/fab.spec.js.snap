// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab Debe hacer match con el snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
